toolchain/check/eval.cpp

```diff
@@ -553,6 +573,62 @@ static auto PerformBuiltinIntComparison(Context& context,
   return MakeBoolResult(context, bool_type_id, result);
 }
 
+// Performs a builtin unary float -> float operation.
+static auto PerformBuiltinUnaryFloatOp(Context& context,
+                                       SemIR::BuiltinFunctionKind builtin_kind,
+                                       SemIR::InstId arg_id)
+    -> SemIR::ConstantId {
+  auto op = context.insts().GetAs<SemIR::FloatLiteral>(arg_id);
+  auto op_val = context.floats().Get(op.float_id);
+
+  switch (builtin_kind) {
+    case SemIR::BuiltinFunctionKind::FloatNegate:
+      op_val = llvm::APFloat(0.0) - op_val;
+      break;
+    default:
+      CARBON_FATAL() << "Unexpected builtin kind";
+  }
+
+  return MakeFloatResult(context, op.type_id, std::move(op_val));
+}
+
+// Performs a builtin binary float -> float operation.
+static auto PerformBuiltinBinaryFloatOp(Context& context, SemIRLoc loc,
+                                        SemIR::BuiltinFunctionKind builtin_kind,
+                                        SemIR::InstId lhs_id,
+                                        SemIR::InstId rhs_id)
+    -> SemIR::ConstantId {
+  auto lhs = context.insts().GetAs<SemIR::FloatLiteral>(lhs_id);
+  auto rhs = context.insts().GetAs<SemIR::FloatLiteral>(rhs_id);
+  auto lhs_val = context.floats().Get(lhs.float_id);
+  auto rhs_val = context.floats().Get(rhs.float_id);
+
+  llvm::APFloat result_val(0.0);
```

Comment:
```suggestion
  llvm::APFloat result_val(lhs_val.getSemantics());
```
to avoid assuming IEEE-754 64-bit.

Reply:
Resolved!